### PR TITLE
mold: Add README

### PR DESCRIPTION
**Pull request by perseoGI**
### Summary
Changes to recipe:  **mold**

#### Motivation
Add mold README

---
- [x] Read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md)
- [x] Checked that this PR is not a duplicate: [list of PRs by recipe](https://github.com/conan-io/conan-center-index/discussions/24240)
- [ ] Tested locally with at least one configuration using a recent version of Conan

### CODE REVIEW

File: recipes/mold/README.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# mold: A Modern Linker
+
+mold is a faster drop-in replacement for existing Unix linkers. It is several
+times quicker than the LLVM lld linker, the second-fastest open-source linker,
+which I initially developed a few years ago. mold aims to enhance developer
+productivity by minimizing build time, particularly in rapid debug-edit-rebuild
+cycles.
```
Comment (jcar87) [inline recipes/mold/README.md:7]:
```suggestion
mold is a faster drop-in replacement for existing Unix linkers. It is several
times quicker than the LLVM lld linker, the second-fastest open-source linker.
mold aims to enhance developer productivity by minimizing build time, 
particularly in rapid debug-edit-rebuild cycles.
```

File: recipes/mold/README.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+You can configure [Conan](https://github.com/conan-io) to download the latest
+version of `mold` and use it as the linker when building your dependencies and
+projects from source.
```
Comment (jcar87) [inline recipes/mold/README.md:11]:
```suggestion
You can configure Conan to download the latest version of `mold` and use it as the linker 
when building your dependencies and projects from source. Currently only supported
when targeting Linux as the platform.
```

File: recipes/mold/README.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+Add the following section to your _host_ profile targeting Linux:
```
Comment (jcar87) [inline recipes/mold/README.md:13]:
```suggestion
To use mold automatically as the linker, you can add the following section to your
_host_ profile that targets Linux. When using gcc, please note that the following
flags require gcc 12.1 or greater.
```

File: recipes/mold/README.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+```
+[tool_requires]
+*:mold/[*]
+[conf]
+# The following config will only work with clang or gcc >= 12
```
Comment (jcar87) [inline recipes/mold/README.md:19]:
```suggestion
```

**Comment by jcar87** [comment]
Side note: we need to double check, but technically we _could_ make it work with gcc "older" than 12.1 - we would need the recipe's package_info to either pass `-B <package_folder>/libexec/bin` as a compiler flag - or consider the `GCC_EXEC_PREFIX` environment variable. It would work, assuming there's an executable called `ld` inside the package (rather than mold), but Im not 100% convinced it's the best course of action (it supplants ld)

**Review by davidsanfal — APPROVED** [review]

**Review by jcar87 — APPROVED** [review]

**Pull request merged**